packages/react-button/src/utils/useToggleState.ts

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { useControllableState } from '@fluentui/react-utilities';
+
+type ToggleProps = {
+  checked?: boolean;
+  defaultChecked?: boolean;
+  disabled?: boolean;
+  disabledFocusable?: boolean;
+  role?: React.AriaRole;
+};
+
+type ToggleState = {
+  'aria-checked'?: React.AriaAttributes['aria-checked'];
+  'aria-pressed'?: React.AriaAttributes['aria-pressed'];
+  checked: boolean;
+  onClick: React.MouseEventHandler<HTMLAnchorElement | HTMLButtonElement>;
+};
+
+export function useToggleState<TToggleButtonProps extends ToggleProps>(props: TToggleButtonProps): ToggleState {
```

Comment:
might be worth using `@testing-library/react-hooks` to test this hook since we expose it in our public API

Reply:
Done.